methods.rb

```diff
@@ -0,0 +1,96 @@
+module Enumerable
+  def my_each
+    for i in self
+      yield(i) if block_given?
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    for j in self
+      yield(j, i) if block_given?
+      i += 1
+    end
+  end
+
+  def my_select
+    result = []
+    self.my_each do |i|
+      result.<<(i) if yield(i)
+    end
+    result
+  end
+
+  def my_all?
```

Comment:
Additional Requirements for this method:
it should:
- return true when no block or argument is given and none of the collection members are false or nil
- return true when a class is passed as an argument and all of the collection is a member of such class
- return true when a Regex is passed as an argument and all of the collection matches the Regex
- return true when a pattern other than Regex or a Class is given and if all of the collection matches the pattern